tests/csm/rest/test_iam_users.py

```diff
@@ -4638,3 +4651,348 @@ def test_42272(self):
+        get_user_list = resp_dict["users"]
+        assert_utils.assert_equals(get_user_list[0], marker, "Marker not set"
+                                                             "to in between user")
+        self.log.info("##### Test completed -  %s #####", test_case_name)
+
+    @pytest.mark.lc
+    @pytest.mark.csmrest
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-42286')
+    def test_42286(self):
+        """
+        Test that internal user should be visible on the GET IAM user list
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("Fetching internal IAM User")
+        resp_node = self.nd_obj.execute_cmd(cmd=common_cmd.K8S_GET_PODS,
+                                            read_lines=False,
+                                            exc=False)
+        pod_name = self.csm_obj.get_pod_name(resp_node)
+        self.log.info(pod_name)
+        self.nd_obj.execute_cmd(
+            cmd=common_cmd.K8S_CP_TO_LOCAL_CMD.format(
+                pod_name, self.cluster_conf_path, self.csm_copy_path, cons.CORTX_CSM_POD),
+            read_lines=False,
+            exc=False)
+        resp = self.nd_obj.copy_file_to_local(
+            remote_path=self.csm_copy_path, local_path=self.local_csm_path)
+        assert_utils.assert_true(resp[0], resp[1])
+        stream = open(self.local_csm_path, 'r')
+        data = yaml.load(stream, Loader=yaml.Loader)
+        internal_user = data["cortx"]["rgw"]["auth_user"]
+        self.log.info("Step 1: Send get request for fetching iam users list")
+        resp = self.csm_obj.list_iam_users_rgw()
+        assert_utils.assert_equals(resp.status_code, HTTPStatus.OK, "Status check failed")
+        self.log.info("Step 2: Check whether internal IAM User is visible in "
+                      "get response")
+        resp_dict = resp.json()
+        get_user_list = resp_dict["users"]
+        assert_utils.assert_in(internal_user, get_user_list,
+                                           "internal user not found in list")
+        self.log.info("##### Test completed -  %s #####", test_case_name)
+
+    @pytest.mark.lc
+    @pytest.mark.csmrest
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-42287')
+    def test_42287(self):
+        """
+        Test Delete internal IAM user should fail
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        test_cfg = self.csm_conf["test_42287"]
+        resp_error_code = test_cfg["error_code"]
+        resp_msg_id = test_cfg["message_id"]
+        resp_data = self.rest_resp_conf[resp_error_code][resp_msg_id]
+        resp_msg_index = test_cfg["message_index"]
+        msg = resp_data[resp_msg_index]
+        self.log.info("Fetching internal IAM User")
```

Comment:
getting internal user details are getting repeated, can we rather add to helper function in this file itself?

Reply:
Sure, will be taking care of this change as a part of Divya's similar comment above